### PR TITLE
Improve code, remove clone when unnecessary

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @sdroege @EPashkin

### CODE REVIEW

File: src/bin/grid.rs [hunk 1]
```diff
@@ -43,22 +43,22 @@ mod example {
         window.set_application(application);
         let grid: Grid = builder.get_object("grid").expect("Couldn't get grid");
         let button6: Button = builder.get_object("button6").expect("Couldn't get button6");
-        button6.connect_clicked(clone!(grid, button6 => move |_| {
-            let height = grid.get_cell_height(&button6);
+        button6.connect_clicked(clone!(grid => move |button| {
```
Comment (sdroege) [inline src/bin/grid.rs]: Is the button inside the grid? If so then this still leads to a reference cycle (use a weak ref for the grid)

**Comment by sdroege** [comment]
Looks good generally

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks

File: src/bin/grid.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/bin/grid.rs]: What do you mean by weak ref? The `WeakRef` type or something else?

Comment (sdroege) [reply src/bin/grid.rs]: Yes, call `downgrade()` before, pass the `WeakRef` into the closure, call `upgrade()` inside the closure and if it returns `None` simply do nothing

**Comment by sdroege** [comment]
Looks good to me. I'll check later if I find other reference cycles in that code but merge for now :)

**Comment by GuillaumeGomez** [comment]
Thanks for reviews!

**Pull request merged**